src/wizard/obtain_source.jl

```diff
@@ -357,6 +357,49 @@ function get_name_and_version(state::WizardState)
     end
 end
 
+@enum Compilers C=1 Go Rust
+function get_compilers(state::WizardState)
+    while state.compilers === nothing
+        compiler_descriptions = Dict(C => "C/C++/Fortran", Go => "Go", Rust => "Rust")
+        compiler_symbols = Dict(Int(C) => :c, Int(Go) => :go, Int(Rust) => :rust)
+        terminal = TTYTerminal("xterm", state.ins, state.outs, state.outs)
+        result = nothing
+        while true
+            select_menu = MultiSelectMenu([compiler_descriptions[i] for i in instances(Compilers)])
+            select_menu.selected = Set([Int(C)])
+            result = request(terminal,
+                             "Select compilers for the project",
+                             select_menu
+                             )
+            if isempty(result)
+                println("Must select at least one platform")
+            else
+                break
+            end
+        end
+        state.compilers = map(c -> compiler_symbols[c], collect(result))
+    end
+end
+
+function get_preferred_version(state::WizardState, compiler::AbstractString,
+                               available_versions=Vector{Integer})
+    version = nothing
+    minver, maxver = getfield.(extrema(available_versions), :major)
+    while version === nothing #&& !eof(state.ins)
+        msg = "Please enter the preferred $(compiler) version ($(minver)...$(maxver))"
+        version = tryparse(Int, nonempty_line_prompt("$(compiler) version", msg;
```

Comment:
I don't think we should `tryparse(Int, nonempty_line_prompt(...))`; I think we should `VersionNumber(nonempty_line_prompt(...))`; that way, `"9"`, `"9.1"` and `"v9.1"` will all work.  Not that we allow point releases now, but it's better to by general, IMO.